iroh-resolver/src/balanced_tree.rs

```diff
@@ -0,0 +1,388 @@
+use std::collections::VecDeque;
+
+use anyhow::Result;
+use async_stream::try_stream;
+use bytes::{Bytes, BytesMut};
+use cid::Cid;
+use futures::{Stream, StreamExt};
+
+use crate::unixfs::{dag_pb, unixfs_pb, DataType, Node, UnixfsNode};
+use crate::unixfs_builder::encode_unixfs_pb;
+
+// 3                          root
+// 2                                 nextparent0
+// 1             parent0              parent1    ... parent7
+// 0 [0, 1, 2, 3, 4, 5, 6, 7]      [0, ... ]
+//
+
+pub fn stream_balanced_tree(
+    in_stream: impl Stream<Item = std::io::Result<BytesMut>>,
+    max_degrees: usize,
+) -> impl Stream<Item = Result<(Cid, Bytes)>> {
```

Comment:
given we get `BytesMut` we should just pass those through, instead of freezing them

Reply:
Hmmm, but we would really only "pass through" the `UnxifsNode::Raw` chunks as `BytesMut`, the `UnixfsNode::File` nodes would get encoded as `Bytes`. We could get the encoded `UnixfsNode::File` as `Bytes` and turn them into `BytesMut` to match the raw nodes?

Thoughts?

Reply:
I guess leave as is for now, can optimize later